### ISSUE TITLE
"AWS provider credentials not found" when using AWS roles

### ISSUE THREAD

**Issue by rkistner** [serverless/serverless#2355]
# This is a Bug Report
## Description

I'm using multiple AWS accounts with role-based delegation across accounts. My `~/.aws/credentials` file is something like this:

```
[main]
aws_access_key_id = ...
aws_secret_access_key = ...

[subaccount]
role_arn = arn:aws:iam::<my-account>:role/Admin
source_profile = main
```

In serverless 1.0.0-beta2, I could use the profile by setting `export AWS_PROFILE=subaccount`. However, in 1.0.2 this doesn't seem to work anymore, and I just get the error "AWS provider credentials not found."

I've also tried setting the profile in `serverless.yml`, but get the same results:

``` yml
provider:
  name: aws
  profile: subaccount
```

Similar issue reported here: https://github.com/serverless/serverless/issues/1787#issuecomment-252317510
## Additional Data
- **_Serverless Framework Version you're using**_: 1.0.2
- **_Operating System**_: Ubuntu 16.04
- **_Stack Trace**_: nothing useful
- **_Provider Error messages**_:

**Comment by WooDzu** [comment]
This also stopped working for me somewhere between RC2 and 1.0.2:

```yamls
provider:
  name: aws
  profile: subaccount
```

**Comment by WooDzu** [comment]
I figured I had environmental leftover `AWS_PROFILE=dev` which is now overriding `provider.profile.subaccount` where before the `provider.profile.subaccount` had precedence.
Panic 's gone.

**Comment by renanosf** [comment]
> I figured I had environmental leftover AWS_PROFILE=dev

Yeah! You're right, I had to remove AWS_PROFILE from my environment variables to work with profile

**Comment by stevecaldwell77** [comment]
I created a PR to fix this: #2687

**Comment by stevecaldwell77** [comment]
> I created a PR to fix this: #2687

..and then noticed that @rkistner had done so as well: #2373.  Anyone reading this should use his PR.

**Comment by pmuens** [comment]
This should be fixed in the latest release.

Let us know if it's still missing and we'll reopen this issue!
